src/lib.rs

```diff
@@ -1277,6 +1277,44 @@ impl Error {
         self.depth
     }
 
+    /// Inspect the underlying [`io::Error`] if there is one.
+    /// If the underlying error isn't an [`io::Error`], [`None`] is returned.
+    ///
+    /// ```rust,no-run
+    /// use walkdir::{WalkDir};
+    /// use std::io;
+    /// use std::path::Path;
+    ///
+    /// let mut it = WalkDir::new("foo").into_iter();
+    /// loop {
+    ///     let entry = match it.next() {
+    ///         None => break,
+    ///         Some(Ok(entry)) => entry,
+    ///         Some(Err(err)) => {
+    ///             println!("failed to access entry {}", err.path().unwrap_or(Path::new("")).display());
+    ///             if let Some(inner) = err.io_error() {
+    ///                 match inner.kind() {
+    ///                     io::ErrorKind::InvalidData => println!("entry contains invalid data: {}", inner),
+    ///                     io::ErrorKind::PermissionDenied => println!("Missing permission to read entry: {}", inner),
+    ///                     _ => println!("Unexpected error occurred: {}", inner),
+    ///                 }
+    ///             }
+    ///             continue
+    ///         }
+    ///     };
+    ///     println!("{}", entry.path().display());
+    /// }
+    /// ```
+    ///
+    /// [`None`]: https://doc.rust-lang.org/stable/std/option/enum.Option.html#variant.None
+    /// [`io::Error`]: https://doc.rust-lang.org/stable/std/io/struct.Error.html
+    pub fn io_error(&self) -> Option<&io::Error> {
+       match self.inner {
+            ErrorInner::Io { path: _ , ref err } => Some(err),
```

Comment:
You don't need to mention `path` here since you aren't using it. This should work: `ErrorInner::Io { ref err, .. }`.